app/services/send_sign_up_email_confirmation.rb

```diff
@@ -5,14 +5,14 @@ def initialize(user)
     @user = user
   end
 
-  def call(request_id: nil, instructions: nil)
+  def call(request_id: nil, instructions: nil, for_password_reset: false)
     remove_legacy_confirmation_info_on_user
     update_email_address_record
 
-    if user.confirmed?
-      send_confirmation_email(request_id, instructions)
-    else
+    if !user.confirmed? && for_password_reset
```

Comment:
Something about this conditional and these method names doesn't quite line up in my head? But I can't come up with anything better? Maybe we could flip them?

```
if user.confirmed? && !for_password_reset
  send_confimirmation_email
else
  send_unconfirmed_email
end
```
```

Reply:
With the current conditional, I'm trying to express that the ONLY time to call `send_unconfirmed_email` is when a password reset has been requested for an unconfirmed user.  Changing the code to your suggestion, as I see it, would not allow for `send_confirmation_email` to be called during new user creation (where `user.confirmed?` is false)

Perhaps I should have used a better name like `password_reset_requested`.  Then the conditional would read more like I just explained: `if password_reset_requested && !user.confirmed?`

Also, perhaps `send_unconfirmed_email` should be more specifically named to something like `send_pw_request_unconfirmed_user_email` ???

Reply:
I like the `password_reset_requested` idea a lot and also yes, let's rename the send method too